### PR TITLE
Default allow K8s Network Policy tests

### PR DESCRIPTION
**Pull request by nebril**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/3409)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by eloycoto** [comment]
I've just spoke with @nebril and we'll wait to merge this PR until https://github.com/cilium/cilium/pull/3379 is merged

**Comment by nebril** [comment]
test-me-please

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -454,6 +456,97 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 				res.ExpectSuccess("Egress DNS connectivity should be allowed for pod %q", pod)
 			}
 		})
+
+		It("Allows traffic with k8s default-allowingress policy", func() {
+
+			res := kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
```
Comment (eloycoto) [inline test/k8sT/Policies.go]: could you change to `elpers.CurlFail("http://%s/public", clusterIP))`?

+ the other ones added ;-)

**Review by eloycoto — APPROVED** [review]
Minor nitpick around Curfail ;-)

**Comment by nebril** [comment]
test-me-please

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -454,6 +456,97 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 				res.ExpectSuccess("Egress DNS connectivity should be allowed for pod %q", pod)
 			}
 		})
+
+		It("Allows traffic with k8s default-allowingress policy", func() {
```
Comment (manalibhutiyani) [inline test/k8sT/Policies.go]: Nit: default-allowingress -> default-allow ingress

File: test/k8sT/Policies.go [hunk 3]
```diff
@@ -454,6 +456,97 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 4, kubectl)
+			Expect(err).To(BeNil(), "Waiting for endpoint updates on %s", ciliumPod)
+
+			By("Testing if egress policy enforcement is enabled on the endpoint")
+
+			var epList []models.Endpoint
+			err = kubectl.CiliumEndpointsList(ciliumPod).Unmarshal(&epList)
+			Expect(err).To(BeNil(), "Getting a list of endpoints from %s", ciliumPod)
+
+			epsWithEgress := 0
```
Comment (manalibhutiyani) [inline test/k8sT/Policies.go]: Can we not use this instead:

```
endpoints, err := kubectl.CiliumEndpointsListByLabel(ciliumPod, podFilter)
policyStatus := endpoints.GetPolicyStatus()
Expect(policyStatus[models.EndpointPolicyEnabledEgress]).Should(Equal(4))
```

**Review by manalibhutiyani — APPROVED** [review]
looks good. please find comments inline

File: test/k8sT/Policies.go (hunk 3 above)
Comment (nebril) [reply test/k8sT/Policies.go]: I think we can, thanks!

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

File: test/k8sT/Policies.go [hunk 4]
```diff
@@ -457,6 +459,88 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+
+			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 4, kubectl)
+			Expect(err).To(BeNil(), "Waiting for endpoint updates on %s", ciliumPod)
+
+			By("Testing connectivity with ingress default-allow policy loaded")
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("Ingress connectivity should be allowed by policy")
```
Comment (aanm) [inline test/k8sT/Policies.go]: If this should be allowed, shouldn't the `CurlFail` be `Curl` in the previous line?

File: test/k8sT/Policies.go [hunk 5]
```diff
@@ -457,6 +459,88 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("Ingress connectivity should be allowed by policy")
```
Comment (aanm) [inline test/k8sT/Policies.go]: Same here

File: test/k8sT/Policies.go [hunk 6]
```diff
@@ -457,6 +459,88 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+			Expect(err).To(BeNil(), "Getting a list of endpoints from %s", ciliumPod)
+
+			endpoints, err := kubectl.CiliumEndpointsListByLabel(ciliumPod, podFilter)
+			policyStatus := endpoints.GetPolicyStatus()
+			Expect(policyStatus[models.EndpointPolicyEnabledEgress]).Should(Equal(4))
+			for _, pod := range []string{appPods[helpers.App2], appPods[helpers.App3]} {
+				res := kubectl.ExecPodCmd(
+					helpers.DefaultNamespace, pod,
+					helpers.CurlFail("http://www.google.com/"))
+				res.ExpectSuccess("Egress connectivity should be allowed for pod %q", pod)
```
Comment (aanm) [inline test/k8sT/Policies.go]: Same here

File: test/k8sT/manifests/1.7/knp-default-allow-ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+apiVersion: networking.k8s.io/v1
```
Comment (aanm) [inline test/k8sT/manifests/1.7/knp-default-allow-ingress.yaml:1]: You can remove this file. In 1.7, `policyTypes` is ignored by kube-apiserver

**Review by aanm — CHANGES_REQUESTED** [review]

File: test/k8sT/Policies.go (hunk 4 above)
Comment (nebril) [reply test/k8sT/Policies.go]: Good point.

**Comment by nebril** [comment]
test-me-please

**Comment by ianvernon** [comment]
@nebril can you please close this since https://github.com/cilium/cilium/pull/4133 took care of default allow?

File: test/k8sT/Policies.go [hunk 8]
```diff
@@ -453,6 +455,89 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+			res.ExpectSuccess("%q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("%q cannot curl to %q", appPods[helpers.App3], clusterIP)
+
+			By("Installing ingress default-allow")
+
+			eps := kubectl.CiliumEndpointPolicyVersion(ciliumPod)
```
Comment (eloycoto) [inline test/k8sT/Policies.go:473]: Please delete this one, with the use of CiliumPolicyAction should be ok.

File: test/k8sT/Policies.go [hunk 9]
```diff
@@ -453,6 +455,89 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+
+			_, err := kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, allowIngress, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(),
+				"L3 allow-ingress Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
+
+			defer func() {
```
Comment (eloycoto) [inline test/k8sT/Policies.go:480]: Please don't use defer in the functions, the cleanup should happens on AfterEach.

File: test/k8sT/Policies.go [hunk 10]
```diff
@@ -453,6 +455,89 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+				kubectl.Delete(allowIngress).ExpectSuccess(
+					"AllowIngress policy cannot be deleted")
+			}()
+
+			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 4, kubectl)
```
Comment (eloycoto) [inline test/k8sT/Policies.go:485]: These lines should be removed, CiliumPolicyAction take care of that.

File: test/k8sT/Policies.go [hunk 11]
```diff
@@ -453,6 +455,89 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+
+		It("Allows traffic with k8s default-allow egress policy", func() {
+			if helpers.GetCurrentK8SEnv() == "1.7" {
+				log.Info("K8s 1.7 doesn't offer a default allow for egress")
+				return
+			}
+
+			By("Installing egress default-allow")
+
+			eps := kubectl.CiliumEndpointPolicyVersion(ciliumPod)
```
Comment (eloycoto) [inline test/k8sT/Policies.go:509]: Please remove this lines, as the other test.

File: test/k8sT/Policies.go [hunk 12]
```diff
@@ -453,6 +455,89 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+
+			_, err := kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, allowEgress, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(),
+				"L3 allow-egress Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
+
+			defer func() {
```
Comment (eloycoto) [inline test/k8sT/Policies.go:516]: Please remove teh policy on AfterEach.

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Pull request closed without merging**